.NET/Microsoft.Recognizers.Text.DateTime/DateTimeRecognizer.cs

```diff
@@ -1,173 +1,69 @@
-﻿using System;
-
-using Microsoft.Recognizers.Text.DateTime.Chinese;
+﻿using Microsoft.Recognizers.Text.DateTime.Chinese;
 using Microsoft.Recognizers.Text.DateTime.English;
 using Microsoft.Recognizers.Text.DateTime.French;
+using Microsoft.Recognizers.Text.DateTime.German;
 using Microsoft.Recognizers.Text.DateTime.Portuguese;
 using Microsoft.Recognizers.Text.DateTime.Spanish;
-using Microsoft.Recognizers.Text.DateTime.German;
-
-using Microsoft.Recognizers.Text.Utilities;
+using System.Collections.Generic;
 
 namespace Microsoft.Recognizers.Text.DateTime
 {
-    public class DateTimeRecognizer : Recognizer
+    public class DateTimeRecognizer : Recognizer<DateTimeOptions>
     {
-
-        private DateTimeOptions instanceOptions;
-
-        public static DateTimeOptions Convert(int value)
-        {
-            return EnumUtils.Convert<DateTimeOptions>(value);
-        }

```

Comment:
These enum Convert methods cannot be removed. In many cases external users of the recognizers do not have direct visibility of the Enum and need to safe cast and int to it. This behaviour is caused by the fact that these are flagged enums.

Reply:
We included constructor overloads to allow options to be passed as `int` values to the recognizer:
```C#
protected Recognizer(string defaultCulture, int options, bool lazyInitialization = true)
    : this(defaultCulture, EnumUtils.Convert<TModelOptions>(options), lazyInitialization)
{
}
```

Reply:
We create a generic method for convertin `int` to the corresponding options `enum`:
```C#
public static TModelOptions GetOption(int value) => EnumUtils.Convert<TModelOptions>(value);
```